keentools/common/escapers.py

```diff
@@ -31,31 +37,144 @@
 _log = KTLogger(__name__)
 
 
+def fb_pinmode_escaper_check() -> bool:
```

Comment:
seems like 3 copypasted functions. it ll be cool if you replace it with single one with settings arg

Reply:
+, it would be nice to generalize these functions (mb using the "product" parameter, as in the `get_settings` function)